odl/contrib/datasets/ct/mayo.py

```diff
@@ -47,43 +51,48 @@ def _read_projections(folder, indices):
 
     file_names = file_names[indices]
 
-    data_array = None
-
     for i, file_name in enumerate(tqdm.tqdm(file_names,
                                             'Loading projection data')):
         # read the file
-        dataset = dicom.read_file(folder + '/' + file_name)
+        try:
+            dataset = pydicom.read_file(folder + os.path.sep + file_name)
+        except:
+            print("corrupted file: {}".format(file_name), file=sys.stderr)
+            print("error:\n{}".format(sys.exc_info()[1]), file=sys.stderr)
+            print("skipping to next file..", file=sys.stderr)
+            continue
```

Comment:
I don't get this part. Why continue when we fail to read a file? We will have no clue what data will come out at the end.
I'd very much prefer this part to just fail if something cannot be read.